### PR TITLE
Issue 72 dbsetup load dbdump

### PR DESCRIPTION
**Pull request by rbock44**
Adds two new modules

gcloud_dbdump_reader
gcloud_dbdump_writer

which adds a service account to the project to read or write the sql dump file to the bucket gs://anonymous-data in serlo-shared.

The modules are used to make it optional as this cannot work in minikube. Also the reader will not be available in production and the writer will only be available in production and for test purposes in dev.

The setup of the reader was tested successfully in dev cluster. The writer is still under development and is currently not used.

### CODE REVIEW

File: modules/athene2_dbdump/variables.tf [hunk 1]
```diff
@@ -26,3 +26,18 @@ variable database_password_readonly {
...
 }
+
+variable "gcloud_bucket_url" {
+  description = "Bucket URL for anonymous database dump should be set only in gcloud environment"
+  default     = "gs://anonymous-data"
+}
+
+variable "gcloud_service_account_key" {
+  description = "Private key of gcloud service account to access the anonymous dtabase dump"
+  default     = ""
```
Comment (inyono) [inline modules/athene2_dbdump/variables.tf]: Does this work correctly with value ""? If not, we shouldn't provide a default value.

File: modules/athene2_dbdump/variables.tf [hunk 2]
```diff
@@ -26,3 +26,18 @@ variable database_password_readonly {
...
+}
+
+variable "gcloud_service_account_name" {
+  description = "Name of the gcloud service account to access the anonymous database dump"
+  default     = ""
```
Comment (inyono) [inline modules/athene2_dbdump/variables.tf]: same here

File: modules/athene2_dbsetup/variables.tf [hunk 3]
```diff
@@ -25,4 +25,19 @@ variable "database_password_default" {
...
+}
+
+variable "gcloud_bucket_url" {
+  description = "Bucket URL for anonymous database dump should be set only in gcloud environment"
+  default     = "gs://anonymous-data"
+}
+
+variable "gcloud_service_account_key" {
+  description = "Private key of gcloud service account to access the anonymous dtabase dump"
+  default     = ""
```
Comment (inyono) [inline modules/athene2_dbsetup/variables.tf]: same here

File: modules/athene2_dbsetup/variables.tf [hunk 4]
```diff
@@ -25,4 +25,19 @@ variable "database_password_default" {
...
+}
+
+variable "gcloud_service_account_name" {
+  description = "Name of the gcloud service account to access the anonymous database dump"
+  default     = ""
 }
```
Comment (inyono) [inline modules/athene2_dbsetup/variables.tf]: same here

File: modules/athene2_dbdump/variables.tf (hunk 1 above)
Comment (rbock44) [reply modules/athene2_dbdump/variables.tf]: It works with empty key. As in this case no dump is uploaded but the dumper still keeps it local. But I am not using it so we can remove it.

File: modules/athene2_dbdump/variables.tf (hunk 2 above)
Comment (rbock44) [reply modules/athene2_dbdump/variables.tf]: Same as above. Will remove it.

File: modules/athene2_dbsetup/variables.tf (hunk 4 above)
Comment (rbock44) [reply modules/athene2_dbsetup/variables.tf]: Same. Actually it is better to set it outside to "" to indicate that this is on purpose. By default we should provide a valid key and name.

**Comment by inyono** [comment]
applied to staging

**Pull request merged**